lib/backends/wasm/ops/binary-op.ts

```diff
@@ -0,0 +1,73 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+import {BinaryOp} from '../../../ops/binary-op';
+import {Tensor} from '../../../tensor';
+import {BroadcastUtil, ShapeUtil} from '../../../util';
+import {WasmBinding} from '../../../wasm-binding';
+import {WasmInferenceHandler} from '../inference-handler';
+
+export class WasmBinaryOp extends BinaryOp {
+  constructor(typeConstraint: ReadonlyArray<Tensor.DataType>, opType: string, resultType?: Tensor.DataType) {
+    super(typeConstraint, opType, resultType);
+  }
+
+  run(inferenceHandler: WasmInferenceHandler, inputs: Tensor[]): Tensor[] {
+    const outputShape = BroadcastUtil.calcShape(inputs[0].dims, inputs[1].dims, false);
+    if (!outputShape) {
+      throw new Error('not broadcastable');
+    }
+    const resultDataLength = ShapeUtil.size(outputShape);
+    const resultData = new Float32Array(resultDataLength);
+    let fun = '';
+    switch (this.opType) {
+      case 'Add':
+        fun = '_add_f32';
+        break;
+      case 'Sub':
+        fun = '_sub_f32';
+        break;
+      case 'Mul':
+        fun = '_mul_f32';
+        break;
+      case 'Div':
+        fun = '_div_f32';
+        break;
+      case 'Xor':
+        fun = '_xor_f32';
```

Comment:
_xor_f32 doesn't make sense. this operator should only works with a Boolean type